pkg/controller/statusmanager/pod_status.go

```diff
@@ -84,53 +84,93 @@ func SetNodeCondition(conditions *[]corev1.NodeCondition, newCondition corev1.No
+	status.Lock()
+	defer status.Unlock()
+
+	if !*firstBoot {
+		return nil
+	}
+	log.Info("Checking all nsx-node-agent pods for node condition")
+	pods := &corev1.PodList{}
+	err := status.client.List(context.TODO(), pods, client.MatchingLabels{"component": operatortypes.NsxNodeAgentContainerName})
+	if err != nil {
+		log.Error(err, "Error getting pods for node condition")
+		return err
+	}
+	if len(pods.Items) == 0 {
+		log.Info("nsx-node-agent not found for node condition")
+		nodes := corev1.NodeList{}
+		err = status.client.List(context.TODO(), &nodes)
+		if err != nil {
+			log.Error(err, "Failed to get nodes for condition updating")
+			return err
+		}
+		for _, node := range nodes.Items {
+			status.setNodeNetworkUnavailable(node.ObjectMeta.Name, false, nil, "NSXNodeAgent", "Waiting for nsx-node-agent to be created", sharedInfo)
+		}
+		*firstBoot = false
+		return nil
+	}
+	podName := types.NamespacedName{}
+	for _, pod := range pods.Items {
+		status.setNodeConditionFromPod(podName, sharedInfo, &pod)
+	}
+	*firstBoot = false
+	return nil
+}
+
+func (status *StatusManager) setNodeNetworkUnavailable(nodeName string, ready bool, startedAt *time.Time, reason string, message string, sharedInfo *sharedinfo.SharedInfo) {
 	if sharedInfo.LastNetworkAvailable == nil {
 		sharedInfo.LastNetworkAvailable = make(map[string]time.Time)
 	}
+	if ready {
+		log.V(1).Info(fmt.Sprintf("Last network unavailable time of node %s is set to %s", nodeName, *startedAt))
+		sharedInfo.LastNetworkAvailable[nodeName] = *startedAt
+	} else {
+		delete(sharedInfo.LastNetworkAvailable, nodeName)
+	}
 	_, changed := status.combineNode(nodeName, ready, reason, message)
 	if !changed {
 		log.Info("Node condition is not changed")
 		return
 	}
 	log.Info(fmt.Sprintf("Setting status NetworkUnavailable to %v for node %s", !ready, nodeName))
 	if !ready {
-		delete(sharedInfo.LastNetworkAvailable, nodeName)
 		status.updateNodeStatus(nodeName, ready, reason, message)
 		return
 	}
-	last := time.Now()
-	_, present := sharedInfo.LastNetworkAvailable[nodeName]
-	if !present {
-		sharedInfo.LastNetworkAvailable[nodeName] = last
-		log.Info(fmt.Sprintf("Last network unavailable time of node %s is set to %s", nodeName, last))
-
-		go func() {
-			time.Sleep(time.Duration(operatortypes.TimeBeforeRecoverNetwork) * time.Second)
-			if (sharedInfo.LastNetworkAvailable[nodeName]) != last {
-				log.Info(fmt.Sprintf("Last network unavailable time of node %s changed, expecting %v, got %v, goroutine exiting",
-					nodeName, last, sharedInfo.LastNetworkAvailable[nodeName]))
-				return
-			}
-			status.updateNodeStatus(nodeName, ready, reason, message)
+	// When the network status looks ready, we still need to wait for a graceful time
+	now := time.Now()
+	startedTime := now.Sub(*startedAt)
+	go func() {
+		sleepTime := time.Duration(operatortypes.TimeBeforeRecoverNetwork * 1000000000) - startedTime
```

Comment:
You could define `TimeBeforeRecoverNetwork`: 180 * time.Millisecond, then you don't need use `1000000000` here or in line 271

Reply:
OK.. I just thought the const should be simple. done.